### DISCUSSION TITLE
Dead?

### DISCUSSION THREAD

**Discussion by johntdyer** [aqualinkd/AqualinkD#154] [General]
Seems there are almost a dozen open PR's and no commit since 2020... Is this project dead ?

**Comment by ballle98** [comment]
I have a fork where I have merged several of the PRs https://github.com/ballle98/AqualinkD.  Mainly because I have an older pool with a PDA.  It's pretty feature complete.

**Comment by john1158way** [comment]
Shaun has supported me through the issues process this year.    I would not say the project is dead.   If you use it and have a challenge Shaun or the project community will respond to issues.

**Comment by sfeakes** [comment]
It’s not dead.  Due to testing it’s hard for people to give good PRs, some of the open ones will work fine for one or two board versions but fail on others.  So I usually have to pull them manually and include with modifications, other PRs make no sense for this project, or don’t follow how the project is coded so add to support/testing overhead.  But many have already been included, but and I do need to clean them up.

The project supports so many board versions now, it takes a huge amount testing that can’t be automated, and since it’s current functionality includes just about everything possible/ supportable, the updates are around fundamental rs485 buss and timing. I have had the next version ready for close to a year now, but there are a few specific PDA only board versions that have been giving me huge problems, and I haven’t had much time with work commitments.   
There have also been some BS DMCA from Jandys lawyers to deal with over that time, which for obvious reasons I won’t go into on an open forum.

in saying all that, what you will get is something that’s highly stable and probably won’t need many updates.

**Reply by dtpanic** [reply]
Amazing and disappointing that Jandy would harass you for innovating on Aqualink when they haven't for years. They should be paying you, not threatening you. Thanks for everything you've done!

**Reply by johntdyer** [reply]
Is there any channel we can message them to express our outrage over this ?   I totally agree that they should be sponsoring you not threatening!

**Reply by jjarboe01** [reply]
I have gotten all the way to a product support manager with Jandy and been told they are not prioritizing their development on the API's for iAquaLink right now.  Which sucks, because you cannot see your AquaPure data or know that the system is in "General Fault" until you see your pool turn green or check the app every day.  They don't want to innovate, so its time to innovate around them!  I'm looking to bring more ease to this project with Home Assistant and a Docker deployment method for the code by sfeakes (mainly my preference for Docker containers) and finally tweak his code so it works with a wifi RS485 dongle over ethernet.  All of this coming from the BWALink project for Balboa hot tubs that worked incredibly well for me and save hundreds over buying the manufacturer's device.  As far as DMCA, not a lawyer here, but how can they claim DMCA infringement on OPEN-SOURCE projects to make their products better?  I mean, if I were Jandy, I'd be searching the home automation communities and incorporating the features we are building into their products.

**Reply by sfeakes** [reply]
FYI
Docker is already supported, but I’ll be posting an update soon on that to support some new features. Dev is done, just some final testing and documentation needs to be finished.
Home assistant support using their discovery protocol is close to finished and will be in next release.  Support through a custom yaml is already available.

Obviously AqualinkD already tells you the complete detail state of Aquapure SWG

**Reply by jjarboe01** [reply]
I wasn’t aware you have already worked on Docker and HA integrations. I will work on getting the code to work with my Ethernet RS485 dongle and look forward to the future releases!  This will be incredible!  Great work on this, amazing code so far from what I have looked over!  Many many thanks for the work on this!
